nav2_motion_primitives/src/back_up.cpp

```diff
@@ -108,4 +126,55 @@ nav2_tasks::TaskStatus BackUp::controlledBackup()
   return TaskStatus::FAILED;
 }
 
+bool BackUp::pathIsClear()
+{
+  auto robot_pose = std::make_shared<geometry_msgs::msg::PoseWithCovarianceStamped>();
+
+  // TODO(orduno) Obtain the pose from the tf tree. The robot class is currently obtaining the pose
+  //              from amcl, and amcl doesn't update the pose often enough.
+  //              The robot pose might not be updated on sequential backup requests.
+
+  if (!robot_->getCurrentPose(robot_pose)) {
+    if (!initial_pose_received_) {
+      // TODO(orduno) Auto-localization needs to execute the backup primitive even when robot
+      //              has not localized and initial pose has not been received.
+      //              In this case we might want to directly check laser scanner measurements
+      //              for possible obstacles.
+      RCLCPP_WARN(node_->get_logger(), "Backup primitive is not checking for free space."
```

Comment:
This is a motion primitive for multiple users, I dont think there should be any reference to auto-localization which is an independent and up stream item. It breaks the encapsulation of making a general backup maneuver .

Reply:
Agree, will work on a solution for this.

Reply:
I have removed the check for initial robot pose altoghether.

We'll enable a service on the world model that checks for free space relative to the robot using directly the laser scanner. That way even if the robot is not localized, world model can still provide obstacle information around it. Motion primities, since are executing short movements, can use this service.

As you can see, I've created a new branch in origin called `world_model_refactor`, once I finish addressing the feedback on this PR. I'll work on a second PR that creates this new service.